### ISSUE TITLE
VB -> C#: CS0115 no suitable method found to override

### ISSUE THREAD

**Issue by AteCoder** [icsharpcode/CodeConverter#685]
### Input code
```vbnet
Protected Overridable ReadOnly Property AutoClearEnabledQueryCommand As String
```

### Erroneous output
```C#
protected virtual string AutoClearEnabledQueryCommand { get; private set; }
```

### Expected output
```C#
protected virtual string AutoClearEnabledQueryCommand { get; set; }
```

### Details
* Product in use: both
* Version in use: 8.2.1
* Did you see it working in a previous version, which? n/a
* Any other relevant information to the issue, or your interest in contributing a fix. 
Clearly, this seems to expose another conflict between VB and C#, which entails a breaking change.

CS0115 appears on the 'set' element. 

It seems that the C# implementation would work if the VB code property is changed to Read/Write.
```vbnet
Protected Overridable Property AutoClearEnabledQueryCommand As String
```
```C#
protected virtual string AutoClearEnabledQueryCommand { get; set; }
```
If the VB code is not changed before the conversion, the converted code is bound to cause a breaking change.  What is implemented on the converter side, then, is a matter of style and perhaps could be informed by having a set of configuration settings. Such settings could also be used, for example, to address the issue of the resources: replace or add in root.

It would seems if the VB cothat the VB code My take is that the VB code needs to change before conversion

**Comment by GrahamTheCoder** [comment]
I don't think this needs to be a breaking change. The set can be omitted, it's just that we'd need to separately declare a backing field if someone assigns to the backing field of the readonly property, [example](https://sharplab.io/#v2:DYLgbgRgNAJiDUAfACgVwsAlgYwAQGFgBDAZxNwEEBYAKF3t2QCcB7AFwFNtOZcB5MByZNMMIhg64AShyIw+AO2ABPRqwAOQtqoqo2LQrKYBRBeOAcYARVRDl+FgFtHRBbwrkAymxEKA5rgAvLgAREQAjCG0DLie6LgAGgAUAJTRMRn0APq6+oZEJmYS1rZM9k4ubkGhRABMUXQMprxxELTNBMRk6fQ9jOhYeISk5ABCfTEAkgoAFkKYbOTUjZnM7Fw8/ILCohzkMnKKKmosmkzalHoGFgWm5pY2dg7Oru5ePpj+1SEQkX2tiVSE0yuByV3yhXuJSeFVe3wg9T6HVa7SqwzIQA==).

The current addition of a private setter was a workaround to avoid that more verbose output - and to avoid detecting if the _PropertyName syntax was used.

**Comment by AteCoder** [comment]
the backing field is compiler-generated in VB.Net.  Is it generated in C#?  

With this test code:
<
        private string Bar { get; }
        [Fact]
        public void WriteReadOnlyField()
        {
            _Bar = "Bar";
        }
>
I get SCS0103 The name '_Foo' does not exist in the current context.  This is both under .Net 5.0 (C# 9(?)) and Core 3.0.

**Comment by GrahamTheCoder** [comment]
Yes the backing field is generated, but there's no syntax to read write directly. 
But we could explicitly create the field in the conversion with the underscore name

**Comment by AteCoder** [comment]
Per MSDN (reading on stack overflow), the backing field is accessible only from the get and set functions of the property. I guess that is how the automated property is generated.  So indeed, the non-breaking solution would be for the conversion to imitate VB.Net, which exposes the backing property.

May I take this opportunity to thank you, again, for this superb project, which enabled me to both transform my VB code and learn so much about C#. Wishing you a great new year. David

________________________________

**Comment by GrahamTheCoder** [comment]
This might be a bit easier to fix now using [the field keyword](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/field)
